python/cudf/cudf/_lib/pylibcudf/replace.pxd

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
+
+from libcpp cimport bool
+
+from cudf._lib.cpp.replace cimport replace_policy
+
+from .column cimport Column
+from .scalar cimport Scalar
+
+ctypedef fused ReplacementType:
+    Column
+    Scalar
+    replace_policy
+    # Allowing object is a workaround for
+    # https://github.com/cython/cython/issues/5984. See the implementation of
+    # replace_nulls for details.
+    object
+
+
+cpdef Column replace_nulls(Column source_column, ReplacementType replacement)
+
+cpdef Column find_and_replace_all(
+    Column source_column,
+    Column values_to_replace,
+    Column replacement_values,
+)
+
+cpdef Column clamp(
+    Column source_column,
+    Scalar lo,
+    Scalar hi,
+    Scalar lo_replace=*,
```

Comment:
What does this `=*` syntax mean? This doesn't seem like anything I've seen in Python, C++, or Cython before.

Reply:
This is the standard Cython syntax for [default arguments in pxd files](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#optional-arguments). The actual default value must be specified in the implementation (the pyx file) while the declaration just indicates that a default exists so that callers know what valid invocations look like. We have this in a couple of places in our existing Cython like [column.pxd](https://github.com/rapidsai/cudf/blob/branch-24.04/python/cudf/cudf/_lib/column.pxd#L36) and [scalar.pxd](https://github.com/rapidsai/cudf/blob/branch-24.04/python/cudf/cudf/_lib/scalar.pxd#L28)